frontend/server/server.ts

```diff
@@ -88,6 +96,13 @@ const s3Client = new MinioClient({
 /** pod template spec to use for viewer crd */
 const podTemplateSpec = loadJSON(VIEWER_TENSORBOARD_POD_TEMPLATE_SPEC_PATH, k8sHelper.defaultPodTemplateSpec)
 
+/** helper function to retrieve pod logs from argo artifactory. */
+const getPodLogsFromArtifactory = _as_bool(ARGO_ARCHIVE_LOGS) ? k8sHelper.getPodLogsFromArtifactoryHelper(
+  ARGO_ARCHIVE_ARTIFACTORY==='minio' ? minioClient : s3Client,
+  ARGO_ARCHIVE_BUCKETNAME,
```

Comment:
Can the actual log location URL be taken from the Workflow status object?

Reply:
> Just a question on which client to use.

AFAIK, in the Frontend the `s3Client` is actually and instance of `minioClient`. There is some weird logic where `s3` reads data uncompressed while `minio` expects `tar.gz`.

Reply:
Generally, I feel that the UI handling of artifacts can get some improvement. If you see some ways to streamline and improve that part, I hope @jingzhang36 and other people working on UX will be glad to see that.

Reply:
@eterna2 Do you think you can take on this: https://github.com/kubeflow/pipelines/issues/2172 later ?

Reply:
Sure.